src/content/opt-out-ext.js

```diff
@@ -1,174 +1,93 @@
+ * them as they are added
+ * @type {MutationObserver}
  */
-const styleTweet = (element, selectedOption, sliderValue) => {
-  element.classList.remove('opt-out-tw', 'opt-out-tc', 'opt-out-trem');
-
-  const tweetPredictionValue = parseFloat(element.dataset.prediction);
-  const modifyTweetThreshold = parseFloat(sliderValue);
-
-  if (
-    // Tweet is predicted to be misogynist
-    tweetPredictionValue !== 0 &&
-    // Option to modify tweet is turned on
-    modifyTweetThreshold !== 0 &&
-    // Tweets prediction value is enough to modify
-    tweetPredictionValue >= modifyTweetThreshold
-  ) {
-    switch (selectedOption) {
-      case 'text_white':
-        element.classList.add('opt-out-tw');
-        break;
-      case 'text_crossed':
-        element.classList.add('opt-out-tc');
-        break;
-      case 'text_removed':
-        element.classList.add('opt-out-trem');
-        break;
-    }
+const checkTweetListObserver = new MutationObserver(
+  (mutationsList) => {
+    mutationsList.forEach((mutation) => {
+      if (mutation.type === 'childList') {
+        if (mutation.addedNodes.length > 0) {
+          processTweets(tweetSelector, popupPrefs);
+        }
+      }
+    });
   }
-};
+);
 
 /**
- * @description function which calls server for given node, and depending on the response,
- * applies pre-defined action
- * @param node
+ * @description Observer which watches root for addition of TweetList element
+ * @type {MutationObserver}
  */
-const checkText = (node) => {
-  node.classList.add('processing');
-
-  // Get text for req
-  const tweetTextNode = node.querySelector(
-    `${selector} > div ~ div > div ~ div`
-  );
-  const text = tweetTextNode.innerText;
-  const reqBody = { texts: [text] };
-
-  const reqHeaders = new Headers();
-  reqHeaders.set('Content-type', 'application/json;charset=UTF-8');
-
-  fetch(OPT_OUT_API_URL, {
-    method: 'POST',
-    headers: reqHeaders,
-    body: JSON.stringify(reqBody),
-    mode: 'cors'
-  })
-    .then(response => {
-      // If successful response
-      if (response.ok) {
-        // Parse body json
-        response.json().then(body => {
-          const predictions = body.predictions;
-          // If response contains prediction
-          if (predictions && predictions.length > 0) {
-            // Convert prediction state to int
-            const predictionInt = Number(predictions[0]);
-            // Add processing status and prediction to tweet node
-            node.classList.add('processed-true');
-            tweetTextNode.setAttribute(
-              'data-prediction',
-              predictionInt.toString()
-            );
-            styleTweet(tweetTextNode, option, slider);
-          } else {
-            // If no prediction
-            node.classList.add('processed-false');
-          }
-        });
-      } else {
-        console.log('Failed response: ', response);
+const checkTweetListCreation = new MutationObserver(
```

Comment:
Twitter is served as a SPA, so this is not safe: the tweet list element on page 1 may not be the same as the element on page 2, but we'll have already removed our observer.

Reply:
I've fixed this in pr #100.